src/Update/Nix/FetchGit/Utils.hs

```diff
@@ -83,3 +102,21 @@ parseISO8601DateToDay t =
   case parseTimeM False defaultTimeLocale "%Y-%m-%d" justDate of
     Nothing -> Left $ InvalidDateString t
     Just day -> pure day
+
+instance Show Warning where
```

Comment:
yes, the prettyprint `Doc` object is ruining that :(

It's probably best not to have any instance of `Show` on `Warning` because of this.